index.bs

```diff
@@ -365,76 +380,123 @@ spec: webidl
-    permission store.
-    </li>
-  </ol>
-  <p>
-    The steps to <dfn>delete a permission storage entry</dfn> of a
-    <a>permission storage identifier</a> are as follows:
-  </p>
-  <ol>
-    <li>If the <a>user agent</a> has a {{PermissionStorage}} associated
-    with the <a>permission storage identifier</a> in its permission store,
-    it MUST remove it.
-    </li>
-  </ol>
+
+  <section>
+    <h3 id="permission-store-algorithms">Permission store algorithms</h3>
+
+    <p algorithm="read-the-granted-permissions">
+      When an algorithm says to <dfn export local-lt="read">read the granted
+      permissions</dfn> that match a particular description, the UA must return
+      the <a>permission entries</a> that match the description in the <a>current
+      realm's</a> <a>permission store</a> that match the description.
+      <a>Denied flags</a> don't contribute any instances.
+    </p>
+
+    <p algorithm="permission-is-denied">
+      The |name| <dfn export>permission is denied</dfn> if the <a>current
+      realm's</a> <a>permission store</a> holds a <a>denied flag</a> for
+      |name|.
+    </p>
+
+    <p algorithm="write-the-permission-entry">
+      When an algorithm says to <dfn export local-lt="write">write the permission
+      entry</dfn> |entry|, the UA may add |entry| to the <a>current realm's</a>
+      <a>permission store</a>, overwriting any existing <a lt="equivalent
+      permission entry">equivalent</a> entries.
+    </p>
+
+    <p algorithm="revoke-the-permission-entries">
+      When an algorithm says to <dfn export local-lt="revoke">revoke the
+      permission entries</dfn> matching some description, the UA may remove those
+      <a>permission entries</a> from the <a>current realm's</a> <a>permission
+      store</a>.
+    </p>
+
+    <p>
+      There are no algorithms to set the <a>denied flag</a>. It's
+      expected that UAs will set this in response to user action outside the
+      scope of web standards.
+    </p>
+  </section>
+
+  <section>
+    <h3 id="permission-store-constraints">Permission store constraints</h3>
+
+    <p algorithm="relevant-permission-change">
+      A <dfn>relevant permission change</dfn> for one or more <a>realms</a> |realms|
+      consists of either the UA receiving new information about the user's
+      intent or a use of <a>write the permission entry</a> or <a>revoke the
+      permission entries</a> from a <a>realm</a> whose <a
+      lt="the Realm's settings object">settings object</a> has the <a>same
+      origin</a> as one of the |realms|.
+    </p>
+
+    <p class="issue" id="issue-same-domain-permissions">
+      It's not clear if some current browsers share permissions more widely than
+      a single origin. For example, when a user grants permission for
+      https://foo.com/ to use a capability, some browsers may also give access
+      to any origin with a domain ending in ".foo.com". This specification may
+      need to allow this.
+    </p>
```

Comment:
It seems like a very bad idea to cross schemes. Domains might be okay, but that also very much depends.